operators/test/e2e/test/context.go

```diff
@@ -0,0 +1,113 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package test
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"sync"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var (
+	testContextPath = flag.String("testContextPath", "", "Path to the test context file")
+	ctxInit         sync.Once
+	ctx             Context
+	log             = logf.Log.WithName("e2e")
+)
+
+func init() {
+	logf.SetLogger(logf.ZapLogger(true))
+}
+
+// Ctx returns the current test context.
+func Ctx() Context {
+	ctxInit.Do(initializeContext)
+	return ctx
+}
+
+func initializeContext() {
+	if *testContextPath == "" {
```

Comment:
I got a bit confused here: the context is unmarshalled from the context file, which was written from flags when running the tests. Am I correct?
So if I run tests with flags, context args are written into a file then immediately read back from that file to build the actual Context.
Unless invoked in 2 steps: 1. prepare (write the file), 2. run (read the file), which would maybe bring some clarity. The flags would then belong to 2 different subcommands (one for building the context, another one for running the tests).

Reply:
The tests are run by a different process (Kubernetes job executing `go test`). This is why the test context is deserialized from a file. 

There are lots of things that need to happen before the test can be run such as creating the namespaces and deploying the operator. The code I introduced in `cmd/run` does this setup and produces the test context file so that the tests can figure out information pertinent to this test run. (For example, because unique namespaces are created for each test run, there must be a way for the tests to refer to those specific namespaces).